tools/test-framework/src/chain/cli/query.rs

```diff
@@ -120,3 +194,101 @@ pub fn query_cross_chain_query(
 
     Ok(res)
 }
+
+/// Query authority account for a specific module
+pub fn query_auth_module(
+    chain_id: &str,
+    command_path: &str,
+    home_path: &str,
+    rpc_listen_address: &str,
+    module_name: &str,
+) -> Result<String, Error> {
+    let account = match simple_exec(
+        chain_id,
+        command_path,
+        &[
+            "--home",
+            home_path,
+            "--node",
+            rpc_listen_address,
+            "query",
+            "auth",
+            "module-account",
+            module_name,
+            "--output",
+            "json",
+        ],
+    ) {
+        Ok(output) => {
+            let json_res: HashMap<String, serde_json::Value> =
+                serde_json::from_str(&output.stdout).map_err(handle_generic_error)?;
+
+            json_res
+                .get("account")
+                .ok_or_else(|| eyre!("expect `account` string field to be present in json result"))?
+                .clone()
+        }
+        Err(e) => {
+            debug!("CLI `query auth module-account` failed, will try with `query auth module-accounts`: {e}");
+            let output = simple_exec(
+                chain_id,
+                command_path,
+                &[
+                    "--home",
+                    home_path,
+                    "--node",
+                    rpc_listen_address,
+                    "query",
+                    "auth",
+                    "module-accounts",
+                    "--output",
+                    "json",
+                ],
+            )?
+            .stdout;
+            let json_res: HashMap<String, serde_json::Value> =
+                serde_json::from_str(&output).map_err(handle_generic_error)?;
```

Comment:
One thing we could do here and in other places where we deal with JSON to avoid peeking manually into the JSON `Value` is to define a struct matching the JSON we expect, derive `Deserialize` for it and then convert the JSON into the struct.

Can be left for later, or skipped completely, will leave you judge of whether it's worth it or not.

Reply:
I agree, it is a good idea as the amount of methods where we extract the JSON output has increased. I think keeping this in a separate PR to clean the test-framework methods would be better. I will open an issue